### PR TITLE
Add uintptr_t and intptr_t data types (#134)

### PR DESCRIPTION
**Pull request by zeffy**
Since there is no real equivalent to `size_t` in .NET, I hesitated to add it.
The uintptr_t can be used and mapped to size_t if required, however.

I'm not entirely sure if I got everything, as there seems to be a lot of code relating to data types all over the place, but it seems to work for me.

If anything needs to be changed to get this merged in, just let me know and I'll fix it.

### CODE REVIEW

**Comment by zeffy** [comment]
Mmm, seems like there's a minor bug. Hold off on merging until I can fix it (it's not displaying hexadecimal correctly, it shows 0x{base10} instead of base16.

**Comment by zeffy** [comment]
Fixed, IntPtr has its own `ToString(string)` method which can take a format specifier, but UIntPtr does not, so I convert it to UInt32 or UInt64 to format it, depending on if `RECLASS64` is defined. Seemed the most straight forward solution.

File: ReClass.NET/Forms/MainForm.cs [hunk 1]
```diff
@@ -617,6 +619,12 @@ private void searchForEqualValuesToolStripMenuItem_Click(object sender, EventArg
 				case UInt64Node node:
 					comparer = new LongMemoryComparer(ScanCompareType.Equal, (long)node.ReadValueFromMemory(selectedNode.Memory), 0);
 					break;
+				case UIntPtrNode node:
```
Comment (KN4CK3R) [inline ReClass.NET/Forms/MainForm.cs:622]: That's not correct? If you have a x86 process it should be an `IntegerMemoryComparer` and not the 64bit `LongMemoryComparer`.

File: ReClass.NET/Project/CppTypeMapping.cs [hunk 2]
```diff
@@ -6,16 +6,19 @@ namespace ReClassNET.Project
 	public class CppTypeMapping
 	{
 		public string TypeBool { get; set; } = "bool";
+		public string TypeSize { get; set; } = "size_t";
```
Comment (KN4CK3R) [inline ReClass.NET/Project/CppTypeMapping.cs:9]: This type is never used.

**Comment by KN4CK3R** [comment]
Why did you close this PR?

**Pull request closed without merging**